src/core/EntryPoint.sol

```diff
@@ -251,108 +313,55 @@ contract EntryPoint is IEntryPoint, NonceManager, ReentrancyGuard, CallIntentSta
-                if (standard == IIntentStandard(address(0))) {
-                    revert FailedIntent(intentIndex, i, "AA82 unknown standard");
-                }
-            }
-
-            // validate the intent segment itself
-            try standard.validateIntentSegment(intent.intentData[i]) {}
-            catch Error(string memory revertReason) {
-                revert FailedIntent(intentIndex, i, string.concat("AA62 reverted: ", revertReason));
-            } catch {
-                revert FailedIntent(intentIndex, i, "AA62 reverted (or OOG)");
-            }
-        }
-    }
-
     /**
      * Validate user intent.
      * @param intent the user intent to validate.
      * @param intentHash hash of the user's intent data.
      * @param intentIndex the index of this intent.
+     * @param signatureAggregator the allowed signature aggregator.
+     * @param validatedIntents the intents that were validated with the signature aggregator.
      */
-    function _validateUserIntent(UserIntent calldata intent, bytes32 intentHash, uint256 intentIndex)
-        private
-        view
-        returns (uint256 result)
-    {
+    function _validateUserIntentWithAccount(
+        UserIntent calldata intent,
+        bytes32 intentHash,
+        uint256 intentIndex,
+        IAggregator signatureAggregator,
+        bytes32 validatedIntents
+    ) private view {
         // validate intent with account
-        try IAccount(intent.sender).validateUserIntent(intent, intentHash) returns (uint256 _result) {
-            result = _result;
+        try IAccount(intent.sender).validateUserIntent(intent, intentHash) returns (IAggregator aggregator) {
+            //check if intent is to be verified by aggregator
+            if (aggregator != IAggregator(address(0))) {
+                if (aggregator != signatureAggregator) {
+                    revert FailedIntent(
+                        intentIndex, 0, string.concat("AA24 signature error: invalid signature aggregator")
+                    );
+                }
+                if ((uint256(validatedIntents) & (1 << intentIndex)) == 0) {
+                    revert FailedIntent(
+                        intentIndex, 0, string.concat("AA24 signature error: intent not part of aggregate")
+                    );
+                }
+            }
         } catch Error(string memory revertReason) {
-            revert FailedIntent(intentIndex, 0, string.concat("AA23 reverted: ", revertReason));
+            revert FailedIntent(intentIndex, 0, string.concat("AA24 signature error: ", revertReason));
         } catch {
-            revert FailedIntent(intentIndex, 0, "AA23 reverted (or OOG)");
+            revert FailedIntent(intentIndex, 0, "AA24 signature error (or OOG)");
         }
     }
 
     /**
-     * emits an event based on the revert reason
-     */
-    function _emitRevertReason(bytes memory reason, uint256 solIndex) private {
```

Comment:
We no longer log revert reasons as the solver making the transaction should be sophisticated enough to know the transaction won't revert in the first place